modules/administration/pages/custom-ssl.adoc

```diff
@@ -1,97 +1,195 @@
+----
+Ensure that the [systemitem]``set-cname`` parameter is the fully-qualified domain name of your {productname} Server.
+You can use the the [systemitem]``set-cname`` parameter multiple times if you require multiple aliases.
+. Generate the CA certificate RPM:
++
+----
+rhn-ssl-tool --gen-ca --rpm-only --dir="/root/ssl-build" \
+--from-ca-cert=`path_to_certificate_file`
+----
+. Generate the server certificate RPM:
++
+----
+rhn-ssl-tool --gen-server --rpm-only --dir="/root/ssl-build" \
+--from-server-key=`path_to_server_key_file` \
+--from-server-cert=`path_to_server certificate_file`
+----
+You will need to generate a server certificate RPM for each proxy, using their host names and cnames.
+. Create a new CA file that combines the old and new certificate details, and generate a new RPM:
++
+----
+mkdir /root/combined-ssl-build
+cp /root/old-ssl-build/RHN-ORG-TRUSTED-SSL-CERT /root/combined-ssl-build/
+cat /root/ssl-build/RHN-ORG-TRUSTED-SSL-CERT >> /root/combined-ssl-build/RHN-ORG-TRUSTED-SSL-CERT
+cp /root/old-ssl-build/*.rpm /root/combined-ssl-build/
+rhn-ssl-tool --gen-ca --rpm-only --dir="/root/combined-ssl-build"
+----
+// I would like to split up these steps, I think. LKB 2019-09-10
+. Deploy the CA certificate on the server:
++
+----
+/usr/bin/rhn-deploy-ca-cert.pl --source-dir /root/combined-ssl-build \
+--target-dir /srv/www/htdocs/pub/ --trust-dir=/etc/pki/trust/anchors/
+----
+
+When you have completed this procedure, you can deploy the new certificate on your clients.
 
-. Execute [command]``configure-proxy.sh``.
-. When prompted with:
+
+.Procedure: Deploying Certificates on Traditional Clients
+
+. On the client, create a new custom channel using these details:
 +
+* Name: SSL-CA-Channel
+* Label: ssl-ca-channel
+* Parent Channel: <choose the parent channel of a clients>
+* Summary: SSL-CA-Channel
 
++
+For more on creating custom channels, see xref:administration:channel-management.adoc[].
+. Upload the CA certificate RPM to the channel:
++
 ----
-Do you want to import existing certificates?
+rhnpush -c ssl-ca-channel --nosig \
+--ca-chain=/srv/www/htdocs/pub/RHN-ORG-TRUSTED-SSL-CERT \
+/root/combined-ssl-build/rhn-org-trusted-ssl-cert-1.0-2.noarch.rpm
 ----
-+
-Answer with kbd:[y]
-.
-. Continue by following the script prompts.
+. Subscribe all clients to the new ``SSL-CA-Channel`` channel.
+. Install the CA certificate RPM on all clients by updating the channel.
+
+.Procedure: Deploying Certificates on Salt Clients
+
+// Does the above procedure also work for Salt? LKB 2019-09-10
+
+. On the client, in the {productname} {webui}, navigate to menu:Salt[Remote Commands].
+. In the first field, enter [guimenu]``salt-call state.apply certs``.
+In the second field, after the ``@``, enter [guimenu]``*``.
+Click btn:[Find Targets].
+. Check that the appropriate clients are displayed, and click btn:[Run Command] to deploy the certificates on the clients.
+
+image::deploy_cert_salt.png[scaledwidth=80%]
```

Comment:
Hmm, there is the half missing.

- After the old and the new CA is trusted by every client the customer need to replace the server certificate on Server and Proxies.
- test if the connection to the clients are still working
- Deployed the CA file which has only the new Root CA on all clients
- Finally: update the CA Certificate in the Database

Reply:
Is this written up somewhere?

Reply:
Yes, on the same wiki page where you got the other content😉

Look for section "Replace the Server Certificate on the Server and the Proxies" up to the end of the page .https://wiki.microfocus.com/index.php/SUSE_Manager/Replace_CA_Certificate

Reply:
:woman_facepalming: